### PR TITLE
Funclaws

### PR DESCRIPTION
**Pull request by Digdugxx**
Adds 'Funclaws'

A sub-type of Deathclaw which specialises in using the interact verbs on its victim before killing them.

This branch involves moving ERP code from the 'human' mob type to just the 'living' mob type. This of course opens a big can of worms so I would be grateful if any wise coders could point out any big problems this might cause.

Also fixes the perspective in resistance ERP because they never bothered to check when they added it..

### CODE REVIEW

**Comment by Kabramen** [comment]
speedmerge when

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#define CUM_TARGET_THROAT "throat"
+#define CUM_TARGET_VAGINA "vagina"
+#define CUM_TARGET_ANUS "anus"
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]: we didnt undef this, remove this

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+			return
+
+		if(pound_cooldown < world.time)
+			chosen_hole = null
+			while (chosen_hole == null)
+				pickNewHole(M)
+			pound_cooldown = world.time + 50
+
+		pound(M)
+		sleep(rand(1, 3) + 0)
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]: why is there a '+ 0'

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		pound(M)
+		sleep(rand(1, 3) + 0)
+		pound(M)
+
+/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/pickNewHole(var/mob/living/M)
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]:
```suggestion
/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/pickNewHole(mob/living/M)
```

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	switch(rand(2))
+		if(0)
+			chosen_hole = CUM_TARGET_ANUS
+		if(1)
+			if(M.has_vagina())
+				chosen_hole = CUM_TARGET_VAGINA
+		if(2)
+			chosen_hole = CUM_TARGET_THROAT
+
+/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/pound(var/mob/living/M)
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]:
```suggestion
/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/pound(mob/living/M)
```

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+	else if(get_dist(src, M) == 0)
+		if(refactory_period > 0)
+			..()
+			return
+
+		if(pound_cooldown < world.time)
+			chosen_hole = null
+			while (chosen_hole == null)
+				pickNewHole(M)
+			pound_cooldown = world.time + 50
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]: every 5 seconds??

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+				M.reagents.add_reagent("cum", rand(8,12))
+			else
+				message = "cums on \the [M]'s backside."
+		else
+			message = "cums on the floor!"
+
+	playsound(loc, "honk/sound/interactions/clawcum[rand(1, 2)].ogg", 70, 1, -1)
+	visible_message("<font color=purple><b>\The [src]</b> [message]</font>")
+	shake_camera(M, 3, 1)
+	set_is_fucking(null ,null)
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]: ??

Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]: what is this doing here?

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+	refactory_period = 5
+	lust = 5
+	lust_tolerance += 50
+
+	sleep(20)
+	playsound(loc, "honk/sound/interactions/slap.ogg", 70, 1, -1)
+	visible_message("<span class='danger'>\The [src]</b> slaps \the [M] right on the ass!</span>", \
+			"<span class='userdanger'>\The [src]</b> slaps \the [M] right on the ass!</span>", null, COMBAT_MESSAGE_RANGE)
+
+/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/tearSlot(var/mob/living/M, slot)
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]:
```suggestion
/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/tearSlot(mob/living/M, slot)
```

File: code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+/mob/living/simple_animal/hostile/deathclaw/funclaw
+	name = "Funclaw"
+	desc = "A massive, reptilian creature with powerful muscles, razor-sharp claws, and aggression to match. This one seems to have a strange look in its eyes.."
+	var/pound_cooldown = 0
+	var/chosen_hole
+
+
+/mob/living/simple_animal/hostile/deathclaw/funclaw/AttackingTarget()
+	var/mob/living/M = target
+	if(!istype(M, /mob/living/carbon/human) || M.health > 60)
```
Comment (LetterN) [inline code/modules/mob/living/simple_animal/hostile/deathclaw/funclaw.dm]:
```suggestion
	if(!ishuman(M) || M.health > 60)
```

**Comment by MrBobCool** [comment]
This is required to improve player experience.

**Pull request merged**